src/core/etl/src/Flow/ETL/FlowContext.php

```diff
@@ -29,8 +31,10 @@ final class FlowContext
 
     private bool $threadSafe = false;
 
-    public function __construct(public readonly Config $config)
-    {
+    public function __construct(
+        public readonly Config $config,
+        public readonly EntryFactory $entryFactory = new NativeEntryFactory()
```

Comment:
Factory should come from the config and be available through context to create a single configuration point

Reply:
Adjusted.